tests/features/S2N_LIBCRYPTO_SUPPORTS_ENGINE.c

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+/*
+ * This feature probe checks if the linked libcrypto has ENGINE support [1].
+ *
+ * Normally, we would always expect the `openssl/engine.h` header to be
+ * available. However, some platforms (CentOS 10,  Fedora 41, and RHEL 10 are
+ * manually removing the `openssl/engine.h` header (for more detail look at
+ * issues [2] [3]).
+ *
+ * [1] https://docs.openssl.org/1.0.2/man3/engine/
+ * [2] https://github.com/aws/s2n-tls/pull/4705
+ * [3] https://github.com/aws/s2n-tls/pull/4873
+ */
+
+#include <openssl/engine.h>
+
+int main()
+{
+    /* S2N_LIBCRYPTO_SUPPORTS_ENGINE is disabled when compiling with AWS-LC.
+     *
+     * Custom random is only used when linked with OpenSSL. However, it should be
+     * possible to compile the ENGINE related code with other libcrypto (eg.
+     * BoringSSL, LibreSSL). This is not the case with AWS-LC which has a different
+     * `RAND_METHOD` signature and results in a compilation error.
+     *
+     * [1] AWS-LC: https://github.com/aws/aws-lc/blob/main/include/openssl/rand.h#L124
+     * [2] OpenSSL: https://github.com/openssl/openssl/blob/master/include/openssl/rand.h#L42
+     */
+#if defined(OPENSSL_NO_ENGINE) || defined(OPENSSL_IS_AWSLC)
+    #error "engine is not supported"
+#endif
```

Comment:
You're checking for awslc in both the feature probe (compile time) and s2n_supports_custom_rand (runtime). What is the behavior of the engine methods with awslc?